Dockerfile

```diff
@@ -1,5 +1,9 @@
 FROM nginx:1.17.7-alpine
 
+# Fix permission issue
+RUN chmod -R 777 /var/log/nginx /var/cache/nginx/ /var/run/ && \
```

Comment:
Should I leave it as is or use this:

```dockerfile
RUN chown -R 101:0 /var/cache/nginx && \
    chmod -R g+w /var/cache/nginx
```

Reply:
Or this ...

```dockerfile
RUN chown -R nginx:nginx /var/cache/nginx && \
    chmod -R g+w /var/cache/nginx
```

Reply:
So, I'm trying to figure out how best to solve this in a way that we don't need to [give everyone permission to `/var/run`](https://serverfault.com/a/159349).

Ideally, we'd do:
```dockerfile
RUN chown -R nginx:nginx /var/cache/nginx && \
    chmod -R g+w /var/cache/nginx
```

Then in the `/etc/nginx/nginx.conf` file we can set `pid` to `/tmp/nginx.pid`.

In theory we don't need to touch `/var/log/nginx` since the files are already there, writable and symlinked correctly:
```console
/etc/nginx # ls -la /var/log/nginx
total 8
drwxr-xr-x    2 root     root          4096 Jan 23 19:29 .
drwxr-xr-x    1 root     root          4096 Jan 23 19:29 ..
lrwxrwxrwx    1 root     root            11 Jan 23 19:29 access.log -> /dev/stdout
lrwxrwxrwx    1 root     root            11 Jan 23 19:29 error.log -> /dev/stderr
```

Reply:
I agree with your suggestions and we updated it with your suggestions in https://github.com/Intellection/docker-nginx/pull/5/commits/fa00f6dcf114119c1cb0d95fcd47f80c58ee8ac7.